File: 1st sem/2022_01_03/Q4.c
//Q4
//largest array elements

#include<stdio.h>

int large=-32000;
void large_array(int arr[], int start, int length){
    if (start == length)
        return;
    
    if (arr[start]>large)
        large = arr[start];
    
    large_array(arr, ++start, length);
    
}

int main()
{
    int len;
    printf("Enter size of array: ");
    scanf("%d", &len);
    int arr[len];

    for (int i = 0; i < len; i++)
    {
        printf("Enter %d element: ", i+1);
        scanf("%d", &arr[i]);
    }

    large_array(arr, 0, len);
    printf("The largest element is: %d", large);    
    
    return 0;
}

File: 4th sem/WT Lab/2023_02_13/q4.java
import java.util.Scanner;

public class q4 {
    public static void main(String[] args){
        int size;
        System.out.print("Enter size of array: ");
        Scanner sc = new Scanner(System.in);
        size = sc.nextInt();
        int arr[] = new int[size];
        System.out.println("Enter elements of array: ");
        for(int i=0; i<size; i++){
            arr[i] = sc.nextInt();
        }

        int max = arr[0];
        int min = arr[0];
        for(int i=0; i<size; i++){
            if(arr[i] > max){
                max = arr[i];
            }
            if(arr[i] < min){
                min = arr[i];
            }
        }

        System.out.println("Maximum element: "+max);
        System.out.println("Minimum element: "+min);
        sc.close();
    }
}
